From aee5d5bac98a85cacb0eb13c8209b25e366cf41e Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 10 Mar 2014 16:44:21 +0000
Subject: [PATCH 27/28] ensure device_path is null terminated

In the unlikely event that path exceeds the length of device_path
we should null terminate is as strncpy() will not do that for us
with an overly long path

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_kobj_uevent.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/thd_kobj_uevent.cpp b/src/thd_kobj_uevent.cpp
index 3829502..a8de8cf 100644
--- a/src/thd_kobj_uevent.cpp
+++ b/src/thd_kobj_uevent.cpp
@@ -74,4 +74,5 @@ bool cthd_kobj_uevent::check_for_event() {
 
 void cthd_kobj_uevent::register_dev_path(char *path) {
 	strncpy(device_path, path, max_buffer_size);
+	device_path[max_buffer_size - 1] = '\0';
 }
-- 
1.9.0

